src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1723,20 +1723,25 @@ public void interrupt() {
             checkAccess();
         }
 
-        Interruptible blocker;
         synchronized (interruptLock) {
+            // Setting the interrupt status must be done before reading nioBlocker.
             interrupted = true;
             interrupt0();  // inform VM of interrupt
+        }
 
-            // thread may be blocked in an I/O operation
-            blocker = nioBlocker;
+        // thread may be blocked in an I/O operation
+        if (this != Thread.currentThread()) {        
+            Interruptible blocker;
+            synchronized (interruptLock) {
+                blocker = nioBlocker;
+                if (blocker != null) {
+                    blocker.interrupt(this);
+                }
+            }
             if (blocker != null) {
-                blocker.interrupt(this);
+                blocker.postInterrupt();
```

Comment:
I think this method (in its entirety) should be identical to upstream.

Reply:
Matched `Thread.interrupt()` to the upstream, also `Thread.getAndClearInterrupt()`.